dashboard/src/actions/datasetListActions.js

```diff
@@ -1,27 +1,76 @@
 import * as TYPES from "./types";
 
 import API from "../utils/axiosInstance";
+import { DANGER } from "assets/constants/toastConstants";
+import { showToast } from "./toastActions";
 import { uriTemplate } from "utils/helper";
 
-export const fetchPublicDatasets = () => async (dispatch, getState) => {
+export const fetchPublicDatasets = (page) => async (dispatch, getState) => {
   try {
     dispatch({ type: TYPES.LOADING });
     const endpoints = getState().apiEndpoint.endpoints;
+    const { offset, limit, filter, searchKey, perPage } =
+      getState().datasetlist;
+    let publicData = [...getState().datasetlist.publicData];
+    const params = new URLSearchParams();
+    params.append("metadata", "dataset.uploaded");
+    params.append("access", "public");
+    params.append("offset", offset);
+    params.append("limit", limit);
+
+    if (searchKey) {
+      params.append("name", searchKey);
+    }
```

Comment:
`name` is one of the original filters; I didn't remove them when I added the more general `filter`. `?name=foo` is the same as `?filter=dataset.name:~foo`.

Reply:
Hmmm...is `name` going to go away?...or, should we update the API documentation to include it?...or, should it be a "hidden" feature?

Reply:
No reason for `name` to go away; `?name=string` is arguably a bit easier (and certainly more compact) than the modern `?filter=dataset.name:~string`.

It, like all the other "legacy" filters, might as well remain since they require less detailed schema knowledge. But, yeah, somehow `name` (and so far as I can see *only* `name`) got missed in the API documentation page for `GET /datasets`. Huh. 😦